packages/host/app/components/operator-mode.gts

```diff
@@ -0,0 +1,166 @@
+import { TrackedWeakMap } from 'tracked-built-ins';
+import { Model as CardRouteModel } from '../routes/card';
+
+interface Signature {
+  Args: {
+    firstCardInStack: CardRouteModel;
+  };
+}
+
+export default class OperatorMode extends Component<Signature> {
+  @tracked stack: Card[] = [];
+  formats: WeakMap<Card, Format> = new TrackedWeakMap<Card, Format>();
+
+  //A variable to store value of card field
+  //before in edit mode.
+  cardFieldValues: WeakMap<Card, Map<string, any>> = new WeakMap<Card, Map<string, any>>();
+  @service declare cardService: CardService;
+  @service declare router: RouterService;
+
+  constructor(owner: unknown, args: any) {
+    super(owner, args);
+    this.stack = [this.args.firstCardInStack.card!];
+  }
+
+  @action async edit(card: Card) {
+    await this.saveCardFieldValues(card);
+    this.formats.set(card, 'edit');
+  }
+
+  @action getFormat(card: Card): Format | undefined {
+    return this.formats.get(card);
+  }
+
+  @action async close(card: Card) {
+    await this.rollbackCardFieldValue(card);
+    let index = this.stack.indexOf(card);
+    this.stack.splice(index);
+    this.stack = this.stack;
+
+    if (this.stack.length <= 0) {
+      this.router.transitionTo('card', this.args.firstCardInStack);
+    }
+  }
+
+  @action async cancel(card: Card) {
+    await this.rollbackCardFieldValue(card);
+    this.formats.set(card, 'isolated');
+  }
+
+  @action async save(card: Card) {
+    await this.saveCardFieldValues(card);
+    this.write.perform(card);
+  }
+
+  private write = restartableTask(async (card: Card) => {
+    let updatedCard = await this.cardService.saveModel(card);
+    this.formats.set(updatedCard, 'isolated');
+  });
+
+  private async saveCardFieldValues(card: Card) {
+    let fields = await this.cardService.getFields(card);
+    for(let fieldName of Object.keys(fields)) {
+      if (fieldName === 'id') continue;
+
+      let field = fields[fieldName];
+      let fieldCard = field.card as unknown as Card;
+      if ((field.fieldType === 'contains' ||  field.fieldType === 'containsMany') && !(await this.cardService.isPrimitive(fieldCard))) {
+        await this.saveCardFieldValues(fieldCard);
+      }
+
+      let cardFieldValue = this.cardFieldValues.get(card);
+      if (!cardFieldValue) {
+        cardFieldValue = new Map<string, any>;
+      }
+      cardFieldValue.set(fieldName, (card as CardInstanceType<any>)[fieldName]);
+      this.cardFieldValues.set(card, cardFieldValue);
+    }
+  }
+
+  private async rollbackCardFieldValue(card: Card) {
+    let fields = await this.cardService.getFields(card);
+    for(let fieldName of Object.keys(fields)) {
+      if (fieldName === 'id') continue;
+
+      let field = fields[fieldName];
+      let fieldCard = field.card as unknown as Card;
+      if ((field.fieldType === 'contains' ||  field.fieldType === 'containsMany') && !(await this.cardService.isPrimitive(fieldCard))) {
+        await this.rollbackCardFieldValue(fieldCard);
```

Comment:
this is the wrong this to pass in here, you are passing in a card definition. What you want to pass in is the card instance. `card[fieldName]`